src/main/java/com/bal/asha/nurture/server/app/user/service/UserDetailService.java

```diff
@@ -17,11 +18,38 @@ public class UserDetailService {
     private UserDetailRepository userDetailRepository;
 
     public UserDetail save(UserDetail userdetail) {
-        return userDetailRepository.save(userdetail);
+        UserDetail u;
+
+        if(userDetailRepository.findExistByUserEmail(userdetail.getUserEmail())){
+            System.out.println("In if");
+            u=userDetailRepository.findByUserEmail(userdetail.getUserEmail());;
+        }
+        else{
+            System.out.println("In else");
+            u=userDetailRepository.save(userdetail);
+        }
+        return u;
     }
 
-    public  Set<UserDetailDTO> getAllUserRecord(){
+    public  Set<UserDetail> getAllUserRecord(){
         return userDetailRepository.getAllUserRecord();
     }
 
+//    public  Set<UserDetail> update(UserDetail userdetailupdate){
+//
+//        return userDetailRepository.findByUserEmail(userdetailupdate.getUserEmail());
+//    }
+
+    public UserDetail update(UserDetail userdetailupdate) {
+
+          UserDetail userDetail= userDetailRepository.findByUserEmail(userdetailupdate.getUserEmail());
+          userDetail.setAddress(userdetailupdate.getAddress());
```

Comment:
create update method inside entity and pass all updated values instead of using setter